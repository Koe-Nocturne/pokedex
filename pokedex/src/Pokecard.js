import React from 'react';

class Pokecard extends React.Component {
  render() {
    console.log("3");
    console.log("Pokemon ", this.props.pokemon);
    const { id, name, type, base_experience } = this.props.pokemon;
    const imageURL = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`

    return (
      <div>
        <p>{name}</p>
        <img src={imageURL}></img>

        <p>Type: {type}</p>
        <p>EXP: {base_experience}</p>
      </div>
    )
  };
};
export default Pokecard;