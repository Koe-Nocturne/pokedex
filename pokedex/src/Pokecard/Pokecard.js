import React from 'react';
import ReactDOM from 'react-dom';

class Pokecard extends React.Component {
  render() {
    const { id, name, type, base_experience } = this.props.pokemon;
    const imageURL = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`

    return (
      <div>
        <p>{name}</p>
        <img src={imageURL}></img>

        <p>Type: {type}</p>
        <p>EXP: {base_experience}</p>
      </div>
    )
  };
};
export default Pokecard;