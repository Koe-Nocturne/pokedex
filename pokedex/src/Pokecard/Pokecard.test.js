import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json"
import Pokecard from "./Pokecard";
// import { isMainThread } from "worker_threads";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


it("renders without crashing", function(){
  shallow(<Pokecard pokemon={{ id: 4, name: 'Charmander', type: 'fire', base_experience: 62 }} />);
});

it("matches screenshot.", function(){
  let wrapper = shallow(<Pokecard pokemon={{ id: 4, name: 'Charmander', type: 'fire', base_experience: 62 }}/>);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
