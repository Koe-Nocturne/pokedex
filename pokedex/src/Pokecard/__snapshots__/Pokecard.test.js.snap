// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches screenshot. 1`] = `
<div>
  <p>
    Charmander
  </p>
  <img
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
  />
  <p>
    Type: 
    fire
  </p>
  <p>
    EXP: 
    62
  </p>
</div>
`;
